### ISSUE TITLE
is there a way for consumer to set inputs for rerendering?

### ISSUE THREAD

**Issue by alidcastano** [diegohaz/constate#63]

**Comment by diegohaz** [comment]
React Context has an experimental feature called `observedBits`. You can pass a `calculateObservedBits` in the second parameter of `createContext` and pass the relevant bits in the second parameter of `useContext` (or to the `unstable_observedBits` prop on `Consumer`).

**Comment by alidcastano** [comment]
didn't this lib handle this for the end user in a previous version of constate? I say that because im aware of `observerBits` but imo it'd be preferable to have this handled at the library level

**Comment by diegohaz** [comment]
Yep! The current API is the result of some experimenting with several others around React Hooks. You can find them all here: https://github.com/diegohaz/constate/blob/master/CHANGELOG.md

In a previous version, there was only one Context/Provider for the whole state, so it made sense to limit  re-renders on consumers based on slices of that state (namely context keys).

I don't think it makes too much sense introducing this kind of complexity now that contexts are more granular. Even if it did, I can't think on a way to do that since the state shape is unknown at the time the context is created.
